scripts/submit_form.js

```diff
@@ -0,0 +1,113 @@
+// Google Apps Script to submit a form to a Google Sheet and send an email notification
+
+var SCRIPT_PROPERTIES = PropertiesService.getScriptProperties();
+const EMAIL_ADDRESS = "";  // set this!
+
+const formIdToNameMap = new Map([
+  ["feedback", "Give Feedback"],
+  ["report_missing_info", "Report Missing or Outdated Information"],
+  ["partner_with_freefrom", "Partner with FreeFrom"],
+  ["build_collective_survivor_power", "Build Collective Survivor Power"],
+  ["policy_ideas", "Share your Policy Ideas"],
+]);
+
+function setup() {
+  var doc = SpreadsheetApp.getActiveSpreadsheet();
+  SCRIPT_PROPERTIES.setProperty("key", doc.getId());
+}
+
+function doPost(request) {
+  const { _, postData: { contents, type } = {} } = request;
+
+  if (type !== "application/json") {
+    return ContentService.createTextOutput(
+      JSON.stringify({
+        result: "error",
+        error: "expected type application/json",
+      })
+    ).setMimeType(ContentService.MimeType.JSON);
+  }
+
+  const data = JSON.parse(contents);
+  return submitForm(flattenArrays(data));
+}
+
+// turns arrays into a string of comma-separated values
+function flattenArrays(data) {
+  for (var key in data) {
+    if (Array.isArray(data[key])) {
+      data[key] = data[key].join(", ");
+    }
+  }
+  return data;
+}
+
+function submitForm(data) {
+  var lock = LockService.getScriptLock();
+  lock.waitLock(10000); // wait 10 seconds before conceding defeat
+
+  try {
+    var doc = SpreadsheetApp.openById(SCRIPT_PROPERTIES.getProperty("key"));
+    var formId = data["form"];
+    var formName = formIdToNameMap.get(formId);
+    var sheet = doc.getSheetByName(formName);
+
+    // question keys are on the first row
+    var questions = sheet
+      .getRange(1, 1, 1, sheet.getLastColumn())
+      .getValues()[0];
+
+    // friendly question names are on the second row
+    var questionsFriendly = sheet
+      .getRange(2, 1, 2, sheet.getLastColumn())
+      .getValues()[0];
+
+    // get next empty row
+    var nextRow = sheet.getLastRow() + 1;
+    var row = [];
+    var submitted = { form: formId };
+
+    // loop through the form questions
+    var question, response;
+    for (question of questions) {
+      if (question === "submitted_at") {
+        response = new Date();
+      } else {
+        response = question in data ? data[question] : "";
+      }
+      row.push(response);
+      submitted[question] = response;
+    }
+    sheet.getRange(nextRow, 1, 1, row.length).setValues([row]);
+
+    return ContentService.createTextOutput(
+      JSON.stringify(submitted)
+    ).setMimeType(ContentService.MimeType.JSON);
+  } catch (e) {
+    return ContentService.createTextOutput(
+      JSON.stringify({ result: "error", description: e })
+    ).setMimeType(ContentService.MimeType.JSON);
```

Comment:
Pointing out that if the script has an error, it will still return a 201, just with this error body. I can't seem to find a way to return something more appropriate (like a 500), but I'm planning to handle this in the Flask app so that it always raises a 500 if things go wrong (in either Flask or Google)